src/prommis/solvent_extraction/tests/test_sx_model_dynamic.py

```diff
@@ -0,0 +1,261 @@
+from pyomo.environ import (
+    ConcreteModel,
+    check_optimal_termination,
+    value,
+    TransformationFactory,
+    units,
+)
+
+from idaes.core import FlowDirection, FlowsheetBlock
+from idaes.core.initialization import InitializationStatus
+from idaes.core.initialization.block_triangularization import (
+    BlockTriangularizationInitializer,
+)
+from idaes.core.solvers import get_solver
+from idaes.core.util import DiagnosticsToolbox
+
+import pytest
+
+from prommis.leaching.leach_solution_properties import LeachSolutionParameters
+from prommis.solvent_extraction.ree_og_distribution import REESolExOgParameters
+from prommis.solvent_extraction.ree_aq_distribution import REESolExAqParameters
+from prommis.solvent_extraction.solvent_extraction import SolventExtraction
+
+solver = get_solver()
+
+
+class TestSXmodel:
+    @pytest.fixture(scope="class")
```

Comment:
Don't repeat the code here. You should use the methods I suggested you define in the model above here. Otherwise, there is no test for the actual flowsheet code.

Reply:
Okay I will check

Reply:
So Andrew, I saw the pytest code for the leaching flowsheet, are you asking me to write it in the same format?

Reply:
Yes. The point is to test all the code you have written, and if you duplicate the code all you are testing is the duplicated code. This then means the actual code is untested and can start to drift out of sync.

Reply:
Okay, I will do it